src/scripts/main.js

```diff
@@ -1,11 +1,51 @@
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve, reject) => {
```

Comment:
The promise should have a reject handler to handle any potential errors that might occur during the event listening process.